integrations/parsely.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Integration: Parse.ly
+ *
+ * @package Automattic\VIP\Integrations
+ */
+
+namespace Automattic\VIP\Integrations;
+
+/**
+ * Loads Parse.ly VIP Integration.
+ *
+ * @private
+ */
+class ParselyIntegration extends Integration {
+	/**
+	 * Loads the plugin.
+	 *
+	 * @param array $config Configuration for this integration.
+	 *
+	 * @private
+	 */
+	public function load( array $config ): void {
+		// Do not load plugin if already loaded by customer code.
+		if ( class_exists( 'Parsely\Parsely' ) ) {
+			return;
+		}
+
+		// Activating the integration via setting constant whose implementation is already
+		// handled via Automattic\VIP\WP_Parsely_Integration (ideally we should move
+		// all the implementation here such that there will be only one way of managing
+		// the plugin).
+		define( 'VIP_PARSELY_ENABLED', true );
+	}
+
+	/**
+	 * Adds filter which setup vip config of a site or network site.
+	 *
+	 * @return void
+	 */
+	public function setup_config(): void {
+		add_filter( 'wp_parsely_credentials', function() {
+			if ( is_multisite() ) {
+				return $this->get_vip_config_of_current_network_site();
+			}
+
+			return $this->get_vip_config_of_site();
+		} );
```

Comment:
Do we want to validate that `get_vip_config_of_current_network_site` / `get_vip_config_of_site` return the expected values? Or should we just leave that to the parsely plugin?

One use case we need to figure out: I have the plugin active but not configured (i.e. at the platform-level, we have not provisioned the keys yet so the config does not have site ID / secret). We may still need a way to force off the inputs on the settings page. Currently, this will pass empty values to `wp_parsely_credentials` which will mean that the plugin is not put into "managed" mode and P100 users would be able to edit the site ID and secret themselves.

Reply:
> Do we want to validate that get_vip_config_of_current_network_site / get_vip_config_of_site return the expected values? Or should we just leave that to the parsely plugin?

By validation you mean? currently we do have tests to make sure that we are passing right config

> Currently, this will pass empty values to wp_parsely_credentials which will mean that the plugin is not put into "managed" mode and P100 users would be able to edit the site ID and secret themselves.

Thanks for pointing it out, I was planning to do this work in separate ticket but now will do it in current PR.

Reply:
> Currently, this will pass empty values to wp_parsely_credentials which will mean that the plugin is not put into "managed" mode and P100 users would be able to edit the site ID and secret themselves.

On second thought I think plugin should just check the existance of `wp_parsely_credentials` filter to set the plugin in managed mode (ticket: https://github.com/Automattic/vip-parsely/issues/53)

Let me know if there are any more changes needed here.

Reply:
Added a ticket to add `is_managed` attribute on plugins (https://github.com/orgs/Automattic/projects/611/views/2?pane=issue&itemId=36130791).